### PR TITLE
Changing functionality to write to a json file

### PR DESCRIPTION
**Pull request by mihikabairathi**
Changed the program to parse the config file and store results in a json file instead of printing it out. Also included options in CLI to choose if you want to parse the ruby code or not.

### CODE REVIEW

File: lib/config_parser.rb [hunk 1]
```diff
@@ -2,23 +2,32 @@
...
 class ConfigParser
   def initialize(argv = ARGV)
     @argv = argv
+    @ruby_parse = nil # default - won't parse ruby code in a config file
     prepare_input_parser
     input_validation
     @file_parse = parse_configuration
-    print_configuration(@file_parse)
+    @proto_obj = proto_configuration(@file_parse)
+    File.write('../json/config.json', Config::Directive.encode_json(@proto_obj))
```
Comment (StevenYCChou) [inline lib/config_parser.rb:16]: Just to give @davidbtucker on some context here if you are going to review it - I had an offline discussion with Mihika on where to output data so another process can read it. As this is an file used internally so I think where to put it is up to us. 

I think a special directory `json` is fine for now when Mihika and I discussed it - though I know normally I would expect this file will be written in a runtime environment.

File: lib/config_parser.rb [hunk 2]
```diff
@@ -42,29 +51,29 @@ def parse_configuration
...
+  def proto_configuration(ele_obj)
+    # stores name, attributes, elements of each element of config with proto
+    ele_dir = Config::Directive.new
     # name
-    puts "#{blank}name : #{ele_obj.name}"
-    # attributes
-    ele_obj.each do |a|
-      puts "#{blank}attr #{a[0]} : #{a[1]}"
+    ele_dir.name = ele_obj.name
+    # arguments
```
[the remark this answers is not in the document]
Comment (mihikabairathi) [reply lib/config_parser.rb:66]: Okay, will do that!

File: lib/config_parser.rb [hunk 3]
```diff
@@ -42,29 +51,29 @@ def parse_configuration
...
+    puts 'An error occured while parsing.\n'
     usage(e)
     exit(false)
   end
 
-  def print_configuration(ele_obj, depth = 0)
-    # displays name, attributes, elements of each element in config file
-    blank = ' ' * depth
+  def proto_configuration(ele_obj)
+    # stores name, attributes, elements of each element of config with proto
```
[the remark this answers is not in the document]
Comment (mihikabairathi) [reply lib/config_parser.rb:62]: Sounds good, will make that change everywhere!

File: lib/config_parser.rb [hunk 4]
```diff
@@ -2,23 +2,32 @@
...
 # Accepts a file path and prints out parsed version
 class ConfigParser
   def initialize(argv = ARGV)
     @argv = argv
+    @ruby_parse = nil # default - won't parse ruby code in a config file
     prepare_input_parser
     input_validation
     @file_parse = parse_configuration
-    print_configuration(@file_parse)
+    @proto_obj = proto_configuration(@file_parse)
```
[the remark this answers is not in the document]
Comment (mihikabairathi) [reply lib/config_parser.rb:15]: Awesome, will change that!

File: lib/config_parser.rb [hunk 5]
```diff
@@ -42,29 +51,29 @@ def parse_configuration
...
     # name
-    puts "#{blank}name : #{ele_obj.name}"
-    # attributes
-    ele_obj.each do |a|
-      puts "#{blank}attr #{a[0]} : #{a[1]}"
+    ele_dir.name = ele_obj.name
+    # arguments
+    ele_dir.args = ele_obj.arg
+    # parameters
+    ele_obj.each do |p|
```
[the remark this answers is not in the document]
Comment (mihikabairathi) [reply lib/config_parser.rb:69]: Yep it did! Will keep this change!

**Pull request merged**